src/Lykke.Job.CandlesHistoryWriter.Services/Candles/CandlesCacheInitalizationService.cs

```diff
@@ -47,6 +50,13 @@ public async Task InitializeCacheAsync()
             // Depending on cache invalidation period and on asset pairs amount, there may be a case when
             // the invalidation timer fires before the cache loading has stopped. This will be a signal 
             // to skip timer-based invalidation.
+            // Below we combine two approaches:
+            // - we're exporting a fast signal for any timer-based routine that we have already been initializing the cache;
+            // - and we additionally block all the cache-related operations for other parts of code.
+            // The first is needed to avoid queueing of timer events. If we simply use blocks, we will finally face a problem
+            // when cache initialization may become infinitly repeated. The second is important for avoidining a multi-threaded
+            // write operations to the cache: if we've got a candle update set, we need to await for cache fill completion and
+            // then proceed.
             lock (_initializationStateLocker)
```

Comment:
Do not use lock inside async. It blocks threads from the thread pool and can lead to thread pool thread starvation

Reply:
missed that as was not introduced in that PR. Good catch

Reply:
Fair remark. Will add the task to fix it